src/EventListener/HttpListener.php

```diff
@@ -5,119 +5,139 @@
     {
-        $span = $this->openTracing->getTracer()->getActiveSpan();
+        if ($this->skipRequest($event->getRequest())) {
+            return;
+        }
+
+        $span = $this->tracer->getActiveSpan();
 
         if ($span) {
             $headers = [];
-            $this->openTracing->getTracer()->inject($span->getContext(), Formats\HTTP_HEADERS, $headers);
+            $this->tracer->inject($span->getContext(), Formats\HTTP_HEADERS, $headers);
             $event->getResponse()->headers->add($headers);
         }
     }
 
     /**
-     * @param PostResponseEvent $event
+     * @param TerminateEvent $event
      */
-    public function onKernelTerminate(PostResponseEvent $event): void
+    public function onKernelTerminate(TerminateEvent $event): void
     {
-        $span = $this->openTracing->getTracer()->getActiveSpan();
+        $span = $this->tracer->getActiveSpan();
 
         if ($span) {
             $span->setTag('http.status_code', $event->getResponse()->getStatusCode());
 
             $span->finish();
         }
 
-        $this->openTracing->getTracer()->flush();
+        $this->tracer->flush();
     }
 
     /**
-     * @param GetResponseForExceptionEvent $event
+     * @param ExceptionEvent $event
      */
-    public function onKernelException(GetResponseForExceptionEvent $event): void
+    public function onKernelException(ExceptionEvent $event): void
     {
-        $span = $this->openTracing->getTracer()->getActiveSpan();
+        if ($this->skipRequest($event->getRequest())) {
+            return;
+        }
+
+        $span = $this->tracer->getActiveSpan();
 
         if ($span) {
             if ($event->hasResponse()) {
                 $headers = [];
-                $this->openTracing->getTracer()->inject($span->getContext(), Formats\HTTP_HEADERS, $headers);
+                $this->tracer->inject($span->getContext(), Formats\HTTP_HEADERS, $headers);
                 $event->getResponse()->headers->add($headers);
 
                 $span->setTag('http.status_code', $event->getResponse()->getStatusCode());
             }
 
             $span->log([
                 'error.kind'   => 'Exception',
-                'error.object' => \get_class($event->getException()),
-                'message'      => $event->getException()->getMessage(),
-                'stack'        => $event->getException()->getTraceAsString(),
+                'error.object' => \get_class($event->getThrowable()),
+                'message'      => $event->getThrowable()->getMessage(),
+                'stack'        => $event->getThrowable()->getTraceAsString(),
             ]);
 
             $span->setTag('error', true);
 
             $span->finish();
         }
 
-        $this->openTracing->getTracer()->flush();
+        $this->tracer->flush();
+    }
+
+    private function skipRequest(Request $request): bool
+    {
+        return in_array($request->attributes->get('_route'), $this->skippedRoutes);
```

Comment:
It's not necessary but for better performance (according to some benchmarks) and/or "code cleanness/**purpose**" you should consider using `isset` by application:

1. create associative array in the Extension (eg `array_combine($mergedConfig['http_listener_skipped_routes'], $mergedConfig['http_listener_skipped_routes'])`)
2. set this accos. array to `'open_tracing.http_listener'` (1&2 == "compile time")
3. use `return isset($this->skippedRoutes[$request->attributes->get('_route')]);`

Main point of this is removing _possible duplications_ in route configuration list ...or ... Did you consider this kind of validation in the Extension?

Reply:
valid point, I've used array_flip in the extension